# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBuiltins.cpp
# Commit: 9111720170a2
# Full Hash: 9111720170a283e7c3801856d5c113a2f17caf20
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-05-06 09:50:52
# Regressor Bug: 1642412
# File Overlap Count: 7
# Description:
#   Bug 1762171 - wasm: Fix cranelift build for table optimizations. r=yury
#   
#   This commit adds back the builtins that were removed in
#   bug 1642412.
#   
# ==============================================================================

diff -r 2340f31f2559 -r 9111720170a2 js/src/wasm/WasmBuiltins.cpp
--- a/js/src/wasm/WasmBuiltins.cpp	Thu May 05 21:49:46 2022 +0000
+++ b/js/src/wasm/WasmBuiltins.cpp	Thu May 05 21:49:46 2022 +0000
@@ -228,12 +228,11 @@
     _FailOnNegI32,
     5,
     {_PTR, _I32, _RoN, _I32, _I32, _END}};
-const SymbolicAddressSignature SASigTableGetFunc = {
-    SymbolicAddress::TableGetFunc,
-    _RoN,
-    _FailOnInvalidRef,
-    3,
-    {_PTR, _I32, _I32, _END}};
+const SymbolicAddressSignature SASigTableGet = {SymbolicAddress::TableGet,
+                                                _RoN,
+                                                _FailOnInvalidRef,
+                                                3,
+                                                {_PTR, _I32, _I32, _END}};
 const SymbolicAddressSignature SASigTableGrow = {
     SymbolicAddress::TableGrow,
     _I32,
@@ -246,12 +245,13 @@
     _FailOnNegI32,
     6,
     {_PTR, _I32, _I32, _I32, _I32, _I32, _END}};
-const SymbolicAddressSignature SASigTableSetFunc = {
-    SymbolicAddress::TableSetFunc,
-    _VOID,
-    _FailOnNegI32,
-    4,
-    {_PTR, _I32, _RoN, _I32, _END}};
+const SymbolicAddressSignature SASigTableSet = {SymbolicAddress::TableSet,
+                                                _VOID,
+                                                _FailOnNegI32,
+                                                4,
+                                                {_PTR, _I32, _RoN, _I32, _END}};
+const SymbolicAddressSignature SASigTableSize = {
+    SymbolicAddress::TableSize, _I32, _Infallible, 2, {_PTR, _I32, _END}};
 const SymbolicAddressSignature SASigRefFunc = {
     SymbolicAddress::RefFunc, _RoN, _FailOnInvalidRef, 2, {_PTR, _I32, _END}};
 const SymbolicAddressSignature SASigPreBarrierFiltering = {
@@ -1219,18 +1219,22 @@
       *abiType = Args_Int32_GeneralInt32Int32Int32Int32Int32;
       MOZ_ASSERT(*abiType == ToABIType(SASigTableInit));
       return FuncCast(Instance::tableInit, *abiType);
-    case SymbolicAddress::TableGetFunc:
+    case SymbolicAddress::TableGet:
       *abiType = Args_General_GeneralInt32Int32;
-      MOZ_ASSERT(*abiType == ToABIType(SASigTableGetFunc));
-      return FuncCast(Instance::tableGetFunc, *abiType);
+      MOZ_ASSERT(*abiType == ToABIType(SASigTableGet));
+      return FuncCast(Instance::tableGet, *abiType);
     case SymbolicAddress::TableGrow:
       *abiType = Args_Int32_GeneralGeneralInt32Int32;
       MOZ_ASSERT(*abiType == ToABIType(SASigTableGrow));
       return FuncCast(Instance::tableGrow, *abiType);
-    case SymbolicAddress::TableSetFunc:
+    case SymbolicAddress::TableSet:
       *abiType = Args_Int32_GeneralInt32GeneralInt32;
-      MOZ_ASSERT(*abiType == ToABIType(SASigTableSetFunc));
-      return FuncCast(Instance::tableSetFunc, *abiType);
+      MOZ_ASSERT(*abiType == ToABIType(SASigTableSet));
+      return FuncCast(Instance::tableSet, *abiType);
+    case SymbolicAddress::TableSize:
+      *abiType = Args_Int32_GeneralInt32;
+      MOZ_ASSERT(*abiType == ToABIType(SASigTableSize));
+      return FuncCast(Instance::tableSize, *abiType);
     case SymbolicAddress::RefFunc:
       *abiType = Args_General_GeneralInt32;
       MOZ_ASSERT(*abiType == ToABIType(SASigRefFunc));
@@ -1434,10 +1438,11 @@
     case SymbolicAddress::TableCopy:
     case SymbolicAddress::ElemDrop:
     case SymbolicAddress::TableFill:
-    case SymbolicAddress::TableGetFunc:
+    case SymbolicAddress::TableGet:
     case SymbolicAddress::TableGrow:
     case SymbolicAddress::TableInit:
-    case SymbolicAddress::TableSetFunc:
+    case SymbolicAddress::TableSet:
+    case SymbolicAddress::TableSize:
     case SymbolicAddress::RefFunc:
     case SymbolicAddress::PreBarrierFiltering:
     case SymbolicAddress::PostBarrier: